packages/wasm-crypto/src/rs/vrf.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019-2021 @polkadot/wasm-crypto authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright 2019 Paritytech via https://github.com/paritytech/schnorrkel-js/
+//
+// Originally developed (as a fork) in https://github.com/polkadot-js/schnorrkel-js/
+// which was adpated from the initial https://github.com/paritytech/schnorrkel-js/
+// forked at commit eff430ddc3090f56317c80654208b8298ef7ab3f
+
+use merlin::Transcript;
+use schnorrkel::{
+    signing_context,
+    vrf::{VRFOutput, VRFProof},
+    PublicKey, SecretKey,
+};
+use wasm_bindgen::prelude::*;
+
+/// Size of VRF output, bytes
+pub const VRF_OUTPUT_SIZE: usize = 32;
+
+/// Size of VRF proof, bytes
+pub const VRF_PROOF_SIZE: usize = 64;
+
+/// Size of VRF raw output, bytes
+pub const VRF_RAW_OUTPUT_SIZE: usize = 16;
+
+/// Size of VRF limit, bytes
+pub const VRF_THRESHOLD_SIZE: usize = 16;
+
+/// Run a Random Verifiable Function (VRF) on one single input (message) transcript,
+/// producing the output signature and corresponding short proof.
+///
+/// * secret: UIntArray with 64 element
+/// * context: Arbitrary length UIntArray
+/// * message: Arbitrary length UIntArray
+///
+/// * returned vector is the 32-byte output (signature) and 64-byte proof.
+#[wasm_bindgen]
+pub fn ext_vrf_sign(secret: &[u8], context: &[u8], message: &[u8]) -> Vec<u8> {
+    let keypair = SecretKey::from_ed25519_bytes(secret).unwrap().to_keypair();
+
+    let (io, proof, _) = keypair.vrf_sign(signing_context(context).bytes(message));
+
+    let mut result: [u8; VRF_OUTPUT_SIZE + VRF_PROOF_SIZE] =
+        [0u8; VRF_OUTPUT_SIZE + VRF_PROOF_SIZE];
```

Comment:
```suggestion
    let keypair = SecretKey::from_ed25519_bytes(secret).unwrap().to_keypair();
    let (io, proof, _) = keypair.vrf_sign(signing_context(context).bytes(message));
    let mut result: [u8; VRF_OUTPUT_SIZE + VRF_PROOF_SIZE] =
        [0u8; VRF_OUTPUT_SIZE + VRF_PROOF_SIZE];

```

Generally would prefer the above, i.e variable assignment separated from return, but that is just OCD, will clean this up to just align a bit better with OCD-world on this side.